### ISSUE TITLE
Feedback on committing specs folder

### ISSUE THREAD

**Issue by Tyriar** [withfig/autocomplete#213]
I notice you're committing the compiled `specs` folder, this can be convenient but comes with tradeoffs like commits end up much larger and minor fluctuations in node module versions could change the compiled output. This is what my git index looked like after the PR told me to run `npm run build`:

```
❯ git st
On branch code-autocomplete
Your branch is up to date with 'origin/code-autocomplete'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   specs/brew.js
        modified:   specs/gcloud.js
        modified:   specs/gcloud/access-context-manager.js
        modified:   specs/gcloud/active-directory.js
        modified:   specs/gcloud/ai-platform.js
        modified:   specs/gcloud/anthos.js
        modified:   specs/gcloud/apigee.js
        modified:   specs/gcloud/app.js
        modified:   specs/gcloud/artifacts.js
        modified:   specs/gcloud/asset.js
        modified:   specs/gcloud/auth.js
        modified:   specs/gcloud/bigtable.js
        modified:   specs/gcloud/builds.js
        modified:   specs/gcloud/cheat-sheet.js
        modified:   specs/gcloud/cloud-shell.js
        modified:   specs/gcloud/components.js
        modified:   specs/gcloud/composer.js
        modified:   specs/gcloud/compute.js
        modified:   specs/gcloud/config.js
        modified:   specs/gcloud/container.js
        modified:   specs/gcloud/data-catalog.js
        modified:   specs/gcloud/dataflow.js
        modified:   specs/gcloud/dataproc.js
        modified:   specs/gcloud/datastore.js
        modified:   specs/gcloud/debug.js
        modified:   specs/gcloud/deployment-manager.js
        modified:   specs/gcloud/dns.js
        modified:   specs/gcloud/docker.js
        modified:   specs/gcloud/domains.js
        modified:   specs/gcloud/endpoints.js
        modified:   specs/gcloud/feedback.js
        modified:   specs/gcloud/filestore.js
        modified:   specs/gcloud/firebase.js
        modified:   specs/gcloud/firestore.js
        modified:   specs/gcloud/functions.js
        modified:   specs/gcloud/game.js
        modified:   specs/gcloud/healthcare.js
        modified:   specs/gcloud/help.js
        modified:   specs/gcloud/iam.js
        modified:   specs/gcloud/iap.js
        modified:   specs/gcloud/identity.js
        modified:   specs/gcloud/info.js
        modified:   specs/gcloud/init.js
        modified:   specs/gcloud/iot.js
        modified:   specs/gcloud/kms.js
        modified:   specs/gcloud/logging.js
        modified:   specs/gcloud/meta.js
        modified:   specs/gcloud/ml-engine.js
        modified:   specs/gcloud/ml.js
        modified:   specs/gcloud/monitoring.js
        modified:   specs/gcloud/network-management.js
        modified:   specs/gcloud/organizations.js
        modified:   specs/gcloud/policy-troubleshoot.js
        modified:   specs/gcloud/projects.js
        modified:   specs/gcloud/pubsub.js
        modified:   specs/gcloud/recommender.js
        modified:   specs/gcloud/redis.js
        modified:   specs/gcloud/resource-manager.js
        modified:   specs/gcloud/run.js
        modified:   specs/gcloud/scc.js
        modified:   specs/gcloud/scheduler.js
        modified:   specs/gcloud/secrets.js
        modified:   specs/gcloud/services.js
        modified:   specs/gcloud/source.js
        modified:   specs/gcloud/spanner.js
        modified:   specs/gcloud/sql.js
        modified:   specs/gcloud/survey.js
        modified:   specs/gcloud/tasks.js
        modified:   specs/gcloud/topic.js
        modified:   specs/gcloud/version.js
        modified:   specs/mask.js
        modified:   specs/php.js
        modified:   specs/poetry.js

no changes added to commit (use "git add" and/or "git commit -a")
```

I suggest not committing specs and adding all compiled files to `.gitignore`, then instead building it on `npm run dev` and on `npm postinstall`.

Another way you could improve this is enforce usage of `npm ci` such that `package-lock.json` is used, but you still end up with compiled files being committed.

**Comment by cstrnt** [comment]
Hey there,

thank you very much for creating this issue and giving some improvements.
We also think that this is kinda annoying, and we are working on it. Our plan is to add them to the `.gitignore` and then building them everytime the PR is merged into the master branch! I'll let you know when this is done

**Comment by mattschrage** [comment]
Previous versions of Fig (v1.0.40 and ealier) would download the `/specs` folder directly from Github. As of 1.0.41, we use release artifacts instead. 🙌

Just waiting for the percent of users on the previous build to drop below 10% before removing all the `.js` files from the `/specs` folder!

**Comment by cstrnt** [comment]
closed by #216
